File: StoreClient/src/app/services/ordine/ordine.ts
import { ClassField } from "@angular/compiler";
import { cliente } from "../cliente/cliente";
import { lineaOrdine } from "../lineaOrdine/lineaOrdine";

export class ordine{
    id: number = 0;
    cliente: cliente | null = null;
    data: Date = new Date();
    lineeOrdini: lineaOrdine[] = [];

    constructor(c: cliente, d: Date, lo: lineaOrdine[]){
        this.cliente = c;
        this.data = d;
        this.lineeOrdini = lo;
    }
}

File: StoreServer/src/main/java/com/example/demo/entities/Ordine.java
package com.example.demo.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;
import java.util.Date;
import java.util.List;

@Getter
@Setter
@ToString
@EqualsAndHashCode
@Entity
@Table(name="ordine")
public class Ordine {
    @Id
    @Column(name = "id", nullable = false)
    @GeneratedValue(strategy=GenerationType.AUTO)
    private long id;

    @ManyToOne
    @JoinColumn(name = "cliente")
    private Cliente cliente;

    @Basic
    @Column(name="data")
    @Temporal(TemporalType.TIMESTAMP)
    private Date data;

    @OneToMany(mappedBy = "ordine", cascade = CascadeType.MERGE)
    private List<LineaOrdine> lineeOrdini;

}
